### PR TITLE
map report for multi-exposure IF

### PR DESCRIPTION
**Pull request by myarjunar**
### What does it fix?
* Ticket: #4649 
* Funded by: DFAT
* Description: 

selection_156

### Checklist:
- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Add to the changelog in metadata.txt if it's a new feature
- [ ] Unit test for new code added
- [ ] Request someone to review or test your PR

### CODE REVIEW

File: safe/impact_function/multi_exposure_wrapper.py [hunk 1]
```diff
@@ -282,8 +296,8 @@ def aggregation_summary(self):
         return self._aggregation_summary
 
     @property
-    def analysis_summary(self):
-        """Property for the aggregation summary.
+    def analysis_impacted(self):
```
Comment (Gustry) [inline safe/impact_function/multi_exposure_wrapper.py]: We try to move to the new names: `analysis_summary` https://github.com/inasafe/inasafe/blob/develop/safe/definitions/layer_purposes.py#L134
Why did you change this one?

File: safe/impact_function/multi_exposure_wrapper.py [hunk 2]
```diff
@@ -395,6 +409,24 @@ def current_impact_function(self):
         """
         return self._current_impact_function
 
+    @property
+    def output_layers_ordered(self):
+        """Return the custom order inout from user.
```
Comment (Gustry) [inline safe/impact_function/multi_exposure_wrapper.py]: s/inout/input/g

File: safe/impact_function/multi_exposure_wrapper.py [hunk 3]
```diff
@@ -471,6 +503,9 @@ def _generate_provenance(self):
             provenance_data_store_uri,
             self.datastore.uri_path)
 
+        # Map title
+        set_provenance(self._provenance, provenance_map_title, self.name)
```
Comment (Gustry) [inline safe/impact_function/multi_exposure_wrapper.py]: Side question, is it not already `provenance_impact_function_name`?

File: safe/impact_function/multi_exposure_wrapper.py [hunk 4]
```diff
@@ -876,6 +912,13 @@ def run(self):
 
             analysis_layers.append(impact_function.analysis_impacted)
             aggregation_layers.append(impact_function.aggregation_summary)
+            list_of_analysis_path.append(
+                impact_function.analysis_impacted.source())
+
+        set_provenance(
+            self._provenance,
+            provenance_layer_exposure_summary,
```
Comment (Gustry) [inline safe/impact_function/multi_exposure_wrapper.py]: Maybe better to create its own provenance because this one is a list. Look for `multi` in the provenance definition file.

File: safe/impact_function/multi_exposure_wrapper.py [hunk 5]
```diff
@@ -876,6 +912,13 @@ def run(self):
 
             analysis_layers.append(impact_function.analysis_impacted)
             aggregation_layers.append(impact_function.aggregation_summary)
+            list_of_analysis_path.append(
+                impact_function.analysis_impacted.source())
```
Comment (Gustry) [inline safe/impact_function/multi_exposure_wrapper.py]: Using `source`, we don't know which provider it is (postgres, ogr...), can you switch to https://github.com/inasafe/inasafe/blob/develop/safe/gis/tools.py#L132 and the next function to read it?

File: safe/impact_function/multi_exposure_wrapper.py (hunk 1 above)
Comment (myarjunar) [reply safe/impact_function/multi_exposure_wrapper.py]: I just try to make it matches the `ImpactFunction` , IMHO we should have the same property name for the same purpose, either to change what's in `ImpactFunction` or change what's in `MultiExposureImpactFunction` . no pressure, very flexible :smile:

File: safe/impact_function/multi_exposure_wrapper.py (hunk 3 above)
Comment (myarjunar) [reply safe/impact_function/multi_exposure_wrapper.py]: yes, but I need map title provenance, and I can't think something better than the name to be the map title

File: safe/impact_function/multi_exposure_wrapper.py (hunk 1 above)
Comment (Gustry) [reply safe/impact_function/multi_exposure_wrapper.py]: Agree. `analysis_impacted` was the old name, before the release of 4.0. Better to update `ImpactFunction` I think

File: safe/impact_function/multi_exposure_wrapper.py (hunk 3 above)
Comment (Gustry) [reply safe/impact_function/multi_exposure_wrapper.py]: Ok, no problem, let's do that until they want a better naming

File: safe/impact_function/multi_exposure_wrapper.py (hunk 1 above)
Comment (myarjunar) [reply safe/impact_function/multi_exposure_wrapper.py]: can I do this in another PR?

File: safe/gui/tools/print_report_dialog.py [hunk 6]
```diff
@@ -233,15 +234,21 @@ def print_as_pdf(self):
         report_path = os.path.dirname(impact_layer.source())
 
         # Get the hazard and exposure definition used in current IF
-        hazard = definition(
-            QgsExpressionContextUtils.projectScope().variable(
-                'hazard_keywords__hazard'))
-        exposure = definition(
-            QgsExpressionContextUtils.projectScope().variable(
-                'exposure_keywords__exposure'))
+        hazard = QgsExpressionContextUtils.projectScope().variable(
```
Comment (Gustry) [inline safe/gui/tools/print_report_dialog.py]: Sorry, I just realized.  We want to clean these variables in the project scope.
The end user can't use them out of the box so IMHO it's not good to expose them to public. Can you fetch these keywords with the impact layer?

File: safe/impact_function/multi_exposure_wrapper.py (hunk 1 above)
Comment (Gustry) [reply safe/impact_function/multi_exposure_wrapper.py]: So why not revert back this one?

Comment (Gustry) [reply safe/impact_function/multi_exposure_wrapper.py]: Because you need to use the same property in `ImpactReport` ?

Comment (myarjunar) [reply safe/impact_function/multi_exposure_wrapper.py]: I use this somewhere in the code actually, ok let me revert this back and make some conditional code first

File: safe/gui/tools/print_report_dialog.py (hunk 6 above)
Comment (myarjunar) [reply safe/gui/tools/print_report_dialog.py]: yes of course

File: safe/impact_function/multi_exposure_wrapper.py (hunk 1 above)
Comment (myarjunar) [reply safe/impact_function/multi_exposure_wrapper.py]: > Because you need to use the same property in ImpactReport ? Go for it, it's fine

yes,  you know me so well :smiley:  ok then, it's easier to refactor if we leave it like this

File: safe/report/extractors/composer.py [hunk 7]
```diff
@@ -240,29 +240,40 @@ def qgis_composer_extractor(impact_report, component_metadata):
...
+        layers = [] + impact_report.extra_layers
+
+    add_supplementary_layers = (
+        not impact_report.multi_exposure_impact_function or not (
+            impact_report.multi_exposure_impact_function.output_layers_ordered)
+    )
+    if add_supplementary_layers:
+        # Check show only impact.
+        show_only_impact = setting(
+            'set_show_only_impact_on_report', False, bool)
```
Comment (ismailsunni) [inline safe/report/extractors/composer.py]: We have set the default value in the `default_settings.py`. So, you can remove the default value (False) here.

File: safe/report/extractors/composer.py [hunk 8]
```diff
@@ -240,29 +240,40 @@ def qgis_composer_extractor(impact_report, component_metadata):
...
+            aggregation_layer_id = provenance['aggregation_layer_id']
+            if aggregation_layer_id:
+                aggregation_layer = layer_registry.mapLayers().get(
+                    aggregation_layer_id, None)
+                layers.append(aggregation_layer)
+
+            layers.append(hazard_layer)
+
+        # check hide exposure settings
+        hide_exposure_flag = setting('setHideExposureFlag', False, bool)
```
Comment (ismailsunni) [inline safe/report/extractors/composer.py]: Same comment like the previous one.

**Comment by myarjunar** [comment]
thanks for the review @ismailsunni @Gustry
